engine/access/state_stream/handler.go

```diff
@@ -14,30 +18,107 @@ type Handler struct {
 	chain flow.Chain
 }
 
-// HandlerOption is used to hand over optional constructor parameters
-type HandlerOption func(*Handler)
-
-func NewHandler(api API, chain flow.Chain, options ...HandlerOption) *Handler {
+func NewHandler(api API, chain flow.Chain) *Handler {
 	h := &Handler{
 		api:   api,
 		chain: chain,
 	}
-	for _, opt := range options {
-		opt(h)
-	}
 	return h
 }
 
 func (h *Handler) GetExecutionDataByBlockID(ctx context.Context, request *access.GetExecutionDataByBlockIDRequest) (*access.GetExecutionDataByBlockIDResponse, error) {
 	blockID, err := convert.BlockID(request.GetBlockId())
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.InvalidArgument, "could not convert block ID: %v", err)
 	}
 
 	execData, err := h.api.GetExecutionDataByBlockID(ctx, blockID)
 	if err != nil {
-		return nil, err
+		return nil, rpc.ConvertError(err, "could no get execution data", codes.Internal)
```

Comment:
it would be nice if we could get NotFound for not yet ready blocks

Reply:
it should. `GetExecutionDataByBlockID` will return a `NotFound` status code if either the block or execution data doesn't exist
https://github.com/onflow/flow-go/blob/6397de16ba4323d474ea5a3c6d5d9996ed7a2694/engine/access/state_stream/backend_executiondata.go#L40